backend/src/neighbourhood.rs

```diff
@@ -200,11 +192,11 @@ impl Neighbourhood {
             bail!("No roads inside the boundary");
         }
 
-        if perimeter_roads.is_empty() {
-            // App breaks without perimeter roads: without perimeter roads, there's only one cell,
+        if primary_roads.is_empty() {
+            // App breaks without primary_roads: without primary roads, there's only one cell,
             // so it counts as disconnected (because it doesn't touch a border intersection), and
             // thus we can't calculate shortcuts through it without those intersections.
-            bail!("No perimeter roads");
+            bail!("No primary roads");
```

Comment:
Interesting... so we can't operate on something at all if it's all residential-highway according to OSM. I don't know any use cases for that in the first place, though

Reply:
Oh! I meant to call this out - I actually wasn't sure what the original comment was about, but I'm assuming it applies equally to the new primary-road based regime.

Reply:
Previously, this was kind of impossible unless the user freehand drew a boundary with no roads, or not even a polygon full of roads, and we didn't see any perimeter roads almost parallel to the boundary.

Now, it's defined by primary roads, so you can't do something like this:
![image](https://github.com/user-attachments/assets/466271be-8c1a-40c8-9843-9154c556ca8b)
I've surfaced the error message better. We can think through if it ever makes sense to allow this -- if the user can override the primary/interior roads, then maybe so.